httputil/request_test.go

```diff
@@ -0,0 +1,60 @@
+package httputil
+
+import (
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"testing"
+)
+
+func TestGetAutTypeNoAuthHeaders(t *testing.T) {
+	res := &http.Response{Header: make(http.Header)}
```

Comment:
This initialization could be good to extract to a helper function that, given a `map[string]string` representing the header values, returns a `*http.Response` initialized with those header values. I'm thinking something like:

```go
func httpResponseFromHeaders(headers map[string]string) *http.Response {
        header := make(http.Header, len(headers))
        for k, v := range headers {
                header.Add(k, v)
        }

        return &http.Response{Header: header}
}
```